jetty-security/src/main/java/org/eclipse/jetty/security/PropertyUserStore.java

```diff
@@ -111,18 +117,63 @@ public Path getConfigPath()
 
     /**
      * Set the Config Path from a String reference to a file
-     * @param configFile the config file
+     * @param configFile the config file can a be a file path or a reference to a file within a jar file <code>jar:file:</code>
      */
     public void setConfigPath(String configFile)
     {
         if (configFile == null)
         {
             _configPath = null;
         }
-        else
+        else if (new File( configFile ).exists())
         {
             _configPath = new File(configFile).toPath();
         }
+        if ( !new File( configFile ).exists() && configFile.startsWith( JAR_FILE ))
+        {
+            // format of the url is jar:file:/foo/bar/beer.jar!/mountain_goat/pale_ale.txt
+            // ideally we'd like to extract this to Resource class?
+            try
+            {
+                _configPath = extractPackedFile( configFile );
+            }
+            catch ( IOException e )
+            {
+                throw new RuntimeException( "cannot extract file from url:" + configFile, e );
+            }
+        }
+    }
+
+    private Path extractPackedFile( String configFile )
+        throws IOException
+    {
+
+        int fileIndex = configFile.indexOf( "!" );
+        String filePath = configFile.substring( JAR_FILE.length(), fileIndex );
+        String entryPath = configFile.substring( fileIndex + 1, configFile.length() );
+
+        try (FileInputStream fileInputStream = new FileInputStream( new File( filePath ) ))
+        {
+            ZipInputStream zin = new ZipInputStream( fileInputStream );
+            for ( ZipEntry e; ( e = zin.getNextEntry() ) != null; )
+            {
+                if ( e.getName().equals( entryPath ) )
+                {
+                    Path extractedPath = Files.createTempFile( "users_store", ".tmp" );
+                    byte[] buffer = new byte[1024];
+                    int bytesRead;
+                    try (OutputStream textOutputStream = Files.newOutputStream( extractedPath ))
+                    {
+                        while ( ( bytesRead = zin.read( buffer ) ) != -1 )
+                        {
+                            textOutputStream.write( buffer, 0, bytesRead );
+                        }
+                    }
+                    return extractedPath;
+                }
+            }
+        }
+        throw new RuntimeException( "cannot find file from url " + configFile );
```

Comment:
Or alternatively unpack the whole jar with 
JarResource r = JarResource.newJarResource(url);
r.copyTo(tmpDir)

If you do it with the ZipInputStream instead, I think you need to close the stream, no?